pkg/controllers/leaderworkerset_controller.go

```diff
@@ -135,6 +138,13 @@ func (r *LeaderWorkerSetReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
+	if !lwsUpdated {
+		// in the cases where lws is updated, the Partition in leaderSts is zero, not a useful event to record
+		if leaderSts != nil && partition != *leaderSts.Spec.UpdateStrategy.RollingUpdate.Partition {
+			r.Record.Eventf(lws, corev1.EventTypeNormal, "GroupsAreUpgrading", fmt.Sprintf("Upgrading replicas %d to %d", *leaderSts.Spec.UpdateStrategy.RollingUpdate.Partition, partition))
```

Comment:
Can we define a constant for `GroupsAreUpgrading` and use it everywhere please. Also, can we change the naming from `upgrade` to `update`? also in the conditions: https://github.com/kubernetes-sigs/lws/blob/main/api/leaderworkerset/v1/leaderworkerset_types.go#L324